### PR TITLE
128973_Generate payment schedule for a downward adjustment

### PR DESCRIPTION
**Pull request by AbidemiAdio**

### CODE REVIEW

File: app/processing/payment/calculate-scheduled-payments/get-settlement-value.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const getSettledValue = (previousSettlements) => {
-  return previousSettlements.reduce((total, settlement) => total + settlement.value, 0)
+  return previousSettlements.reduce((first, settlement) => Math.max(first, settlement.value), Number.NEGATIVE_INFINITY)
```
Comment (marctemp) [inline app/processing/payment/calculate-scheduled-payments/get-settlement-value.js:2]: `first` should be called `maximumValue` or sim. as it's hplding the then `maximumValue` each time

File: test/unit/processing/payment/get-settlement-value.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const getSettledValue = require('../../../../app/processing/payment/calculate-scheduled-payments/get-settlement-value')
+const previousSettlements = JSON.parse(JSON.stringify(require('../../../mock-objects/mock-settlements')))
+describe('get settled value', () => {
```
Comment (marctemp) [inline test/unit/processing/payment/get-settlement-value.test.js:3]: Add whitespace

File: test/unit/processing/payment/get-settlement-value.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  beforeEach(() => {
+
+  })
+
+  test('returns maximum value', () => {
```
Comment (marctemp) [inline test/unit/processing/payment/get-settlement-value.test.js:8]: Add test  for empty array

File: app/processing/payment/calculate-scheduled-payments/get-settlement-value.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 const getSettledValue = (previousSettlements) => {
```
[the remark this answers is not in the document]
Comment (AbidemiAdio) [reply app/processing/payment/calculate-scheduled-payments/get-settlement-value.js:1]: We are returning maximum value because that is the pattern of how we store various payments made, in practice what we want is the amount settled so far, hence reason for leaving the name as it was.

File: test/unit/processing/payment/get-settlement-value.test.js (hunk 3 above)
Comment (AbidemiAdio) [reply test/unit/processing/payment/get-settlement-value.test.js:8]: Good spot for the empty array test.

The best test in this scenario is the behaviour of the final output (schedule object) from combination of various messages from different subscriptions. The test frame-work that the test-team is working on would help with behavioural test, and would also, enable component integration and message injection test, that would help to identify behaviour of  schedule generated.

Comment (marctemp) [reply test/unit/processing/payment/get-settlement-value.test.js:9]: Can we not do something in our integration tests as they are now?

We should have (probs out of scope for this) an integration test from `app/processing/process-statements.js` to test how a statement is generated and sim. to schedule

Comment (AbidemiAdio) [reply test/unit/processing/payment/get-settlement-value.test.js]: Created task for comprehensive integration test for `get-payment-schedule.js `

https://dev.azure.com/defragovuk/DEFRA-FFC/_backlogs/backlog/DEFRA-FFC%20Documents%20Team/Stories/?workitem=182645

File: test/unit/processing/payment/get-settled-value.test.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+const getSettledValue = require('../../../../app/processing/payment/calculate-scheduled-payments/get-settled-value')
+const previousSettlements = JSON.parse(JSON.stringify(require('../../../mock-objects/mock-settlements')))
+
+describe('get settled value', () => {
+  beforeEach(() => {
+
```
[the remark this answers is not in the document]
Comment (AbidemiAdio) [reply test/unit/processing/payment/get-settled-value.test.js]: https://github.com/DEFRA/ffc-pay-statement-constructor/blob/128973/Generate-payment-schedule-for-a-downward-adjustment/test/unit/processing/payment/get-settled-value.test.js

**Review by marctemp — APPROVED** [review]
All suggestions implemented
